frontend/src/interfaces/ILeasePayment.ts

```diff
@@ -12,3 +15,25 @@ export interface ILeasePayment {
   note?: string;
   leasePaymentStatusTypeCode: ITypeCode<string>;
 }
+
+export interface IFormLeasePayment
+  extends ExtendOverride<
+    ILeasePayment,
+    {
+      amountPreTax: NumberFieldValue;
+      amountGst: NumberFieldValue;
+      amountTotal: NumberFieldValue;
+    }
+  > {}
+
+export const formLeasePaymentToApiPayment = (
+  formLeasePayment: IFormLeasePayment,
+  gstConstant?: number,
+): ILeasePayment => {
+  return {
+    ...formLeasePayment,
+    amountPreTax: stringToNull(formLeasePayment.amountPreTax),
+    amountGst: stringToNull(formLeasePayment.amountGst),
+    amountTotal: stringToNull(formLeasePayment.amountTotal),
```

Comment:
quick question - we don't convert the amounts to number before we sent to the API? or is the conversion done prior to this?

Reply:
name of the conversion method is non-intuitive perhaps, but the method will:
if empty string, convert to null
if numeric, do not convert